module/log.go

```diff
@@ -60,6 +60,12 @@ type moduleLogger struct {
 // will send log events back to the module's parent (the RDK) via gRPC when
 // possible and to STDOUT when not possible.
 func NewLoggerFromArgs(moduleName string) logging.Logger {
+	// If no `moduleName` was specified, grab it from the environment (will still be empty
+	// string if not specified in environment.)
+	if moduleName == "" {
+		moduleName, _ = os.LookupEnv("VIAM_MODULE_NAME")
+	}
```

Comment:
There were a couple different ways to do this. Both `ModularMain` and `NewModuleFromArgs` call `NewLoggerFromArgs("")`, so this is what I chose to do. Could also pass in `VIAM_MODULE_NAME` to `NewLoggerFromArgs` from both those call sites. Don't want to break any of this module API though for existing modules (e.g. _not_ take in a `moduleName` parameter to this function.)

Reply:
sgtm, maybe add something to the comment saying this will happen?